#! /bin/sh -e

# gdc-array-init.dpatch by Iain Buclaw <ibuclaw@ubuntu.com>
# DP: Fixes error on static initialization of arrays with trailing comma.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- gcc/d/dmd/parse.c	2010-04-09 19:45:12.963853311 +0100
+++ gcc/d/dmd/parse.c	2010-04-02 12:38:39.000000000 +0100
@@ -2593,6 +2593,7 @@
  			{   t = peek(t);
  			    if (t->value != TOKsemicolon &&
  				t->value != TOKcomma &&
+ 				t->value != TOKrbracket &&
  				t->value != TOKrcurly)
  				goto Lexpression;
  			    break;
